ui: update Database page styles to match design system

vladlos opened the pull request:
Style updates of `database-tables` and `grants` tables.
Additional storybook items to display loading, empty and fulfilled states of updated components

Resolves: #45725

Release note(ui): Updated Admin UI database page styles to match design system.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/47753)
<!-- Reviewable:end -->

dhartunian reviewed (changes requested):
Can you list the specific changes that are being made here and screenshot them? 

What are we doing with the commented-out portions of the code? should those be uncommented ideally?

> Reviewed 3 of 10 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47753#-:-M5i6ouNEvIx4L-IF-0E:b-5i08id)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @elkmaster)

___
*[pkg/ui/src/views/databases/containers/databaseGrants/index.tsx, line 20 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47753#-M5i6ZXSCPety90vxm6s:-M5i6ZXSCPety90vxm6t:b-arhy60) ([raw file](https://github.com/cockroachdb/cockroach/blob/fe361d153f95b78b3b5597236c43390c791eee2a/pkg/ui/src/views/databases/containers/databaseGrants/index.tsx#L20)):*
> ```XML
> import { SortSetting } from "src/views/shared/components/sortabletable";
> import { SortedTable } from "src/views/shared/components/sortedtable";
> import { SummaryCard } from "oss/src/views/shared/components/summaryCard";
> ```

remove `oss` imports
___
*[pkg/ui/src/views/databases/containers/databaseGrants/index.tsx, line 74 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47753#-M5wYkxvE3wCiNhvTM9R:-M5wYkxvE3wCiNhvTM9S:b27xc8m) ([raw file](https://github.com/cockroachdb/cockroach/blob/937a9c91a77cc06b342bbd5b07a4289d74f47dd5/pkg/ui/src/views/databases/containers/databaseGrants/index.tsx#L74)):*
> ```XML
>               <SummaryHeadlineStat
>                 title="Total Users"
>                 tooltip="Total users that have been granted permissions on this table."
> ```

Lost tooltip here as well. Should be restored.
___
*[pkg/ui/src/views/databases/containers/databaseGrants/index.tsx, line 75 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47753#-M5i6bEw9JJb2yMzemMV:-M5i6bEw9JJb2yMzemMW:bs0f4k8) ([raw file](https://github.com/cockroachdb/cockroach/blob/fe361d153f95b78b3b5597236c43390c791eee2a/pkg/ui/src/views/databases/containers/databaseGrants/index.tsx#L75)):*
> ```XML
>               ]}
>               // empty={tableInfos.length}
>               // emptyProps={{
> ```

remove commented code or add "no tables" text/components.
___
*[pkg/ui/src/views/databases/containers/databaseTables/index.tsx, line 151 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47753#-M5wXj0k4oio6snC-scd:-M5wXj0k4oio6snC-sce:b-ixn2x6) ([raw file](https://github.com/cockroachdb/cockroach/blob/937a9c91a77cc06b342bbd5b07a4289d74f47dd5/pkg/ui/src/views/databases/containers/databaseTables/index.tsx#L151)):*
> ```XML
>                 <SummaryHeadlineStat
>                   title="Database Size"
>                   tooltip="Approximate total disk size of this database across all table replicas."
> ```

This change is removing important documentation. Can we preserve the (i) icons with tooltips? they're important.

Can you check w/ Anne if she wanted this box to be reformatted?
___
*[pkg/ui/src/views/shared/util/table.styl, line 70 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47753#-M5i7aGY-qda3R6NHxq8:-M5i7aGY-qda3R6NHxq9:b-b4xs51) ([raw file](https://github.com/cockroachdb/cockroach/blob/fe361d153f95b78b3b5597236c43390c791eee2a/pkg/ui/src/views/shared/util/table.styl#L70)):*
> ```Stylus
> 
>     a
>       display flex
> ```

Same question as `<h2>` below: should we use these tags as containers? I'm a little hesitant especially since it complicates what should be simple css for their display.

I'd rather define a larger `db-name` or `table-name` class and put this svg and img stuff in there. What do you think?
___
*[pkg/ui/styl/base/typography.styl, line 37 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47753#-M5i7KDC7LddqEBcgPLK:-M5i7KDC7LddqEBcgPLL:b-mmuuk0) ([raw file](https://github.com/cockroachdb/cockroach/blob/fe361d153f95b78b3b5597236c43390c791eee2a/pkg/ui/styl/base/typography.styl#L37)):*
> ```Stylus
>   font-family SourceSansPro-Regular
>   img
>     margin-right 11px
> ```

Seems odd to add these to `h2.base-heading`. Are you sure we have no other `<h2>` tags with img elements or others inside?


<!-- Sent from Reviewable.io -->

vladlos reviewed (commented):
redesign of db table list and grants pages:
![Screenshot 2020-04-29 at 18.37.59.png](https://files.reviewableusercontent.io/images/reviewable/12850886/n2Dcw2h5-CMk080jBrKxjC9s/Screenshot%202020-04-29%20at%2018.37.59.png)
![Screenshot 2020-04-29 at 18.38.13.png](https://files.reviewableusercontent.io/images/reviewable/12850886/KJX9f9lX82NFHtenS1RSUZD8/Screenshot%202020-04-29%20at%2018.38.13.png)

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47753#-:-M65c03m7465I5aADk2l:b-dn9shk)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dhartunian)

___
*[pkg/ui/src/views/databases/containers/databaseGrants/index.tsx, line 20 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47753#-M5i6ZXSCPety90vxm6s:-M65awxhCyUGsTYBqdJA:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/fe361d153f95b78b3b5597236c43390c791eee2a/pkg/ui/src/views/databases/containers/databaseGrants/index.tsx#L20)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

remove `oss` imports
</blockquote></details>

Done.
___
*[pkg/ui/src/views/databases/containers/databaseGrants/index.tsx, line 74 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47753#-M5wYkxvE3wCiNhvTM9R:-M65azk2346rGkkljdvy:b-d3bzne) ([raw file](https://github.com/cockroachdb/cockroach/blob/937a9c91a77cc06b342bbd5b07a4289d74f47dd5/pkg/ui/src/views/databases/containers/databaseGrants/index.tsx#L74)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

Lost tooltip here as well. Should be restored.
</blockquote></details>

redesigned `<SummaryHeadlineStat/>` as currently it used only in db pages. returned tooltips
___
*[pkg/ui/src/views/databases/containers/databaseGrants/index.tsx, line 75 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47753#-M5i6bEw9JJb2yMzemMV:-M65ayuI2Ncx8cxUsP3g:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/fe361d153f95b78b3b5597236c43390c791eee2a/pkg/ui/src/views/databases/containers/databaseGrants/index.tsx#L75)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

remove commented code or add "no tables" text/components.
</blockquote></details>

Done.
___
*[pkg/ui/src/views/databases/containers/databaseTables/index.tsx, line 151 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47753#-M5wXj0k4oio6snC-scd:-M65bKqI0b6tSEyAtwjM:b-d3bzne) ([raw file](https://github.com/cockroachdb/cockroach/blob/937a9c91a77cc06b342bbd5b07a4289d74f47dd5/pkg/ui/src/views/databases/containers/databaseTables/index.tsx#L151)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

This change is removing important documentation. Can we preserve the (i) icons with tooltips? they're important.

Can you check w/ Anne if she wanted this box to be reformatted?
</blockquote></details>

redesigned `<SummaryHeadlineStat/>` as currently it used only in db pages. returned tooltips
___
*[pkg/ui/src/views/shared/util/table.styl, line 70 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47753#-M5i7aGY-qda3R6NHxq8:-M65biZa9m2w8iOKOlDJ:b-tcu97e) ([raw file](https://github.com/cockroachdb/cockroach/blob/fe361d153f95b78b3b5597236c43390c791eee2a/pkg/ui/src/views/shared/util/table.styl#L70)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

Same question as `<h2>` below: should we use these tags as containers? I'm a little hesitant especially since it complicates what should be simple css for their display.

I'd rather define a larger `db-name` or `table-name` class and put this svg and img stuff in there. What do you think?
</blockquote></details>

created css class, removed unneeded changes from global styles
___
*[pkg/ui/styl/base/typography.styl, line 37 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47753#-M5i7KDC7LddqEBcgPLK:-M65bd6A7BEAlE8hZDcj:b-ezgbok) ([raw file](https://github.com/cockroachdb/cockroach/blob/fe361d153f95b78b3b5597236c43390c791eee2a/pkg/ui/styl/base/typography.styl#L37)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

Seems odd to add these to `h2.base-heading`. Are you sure we have no other `<h2>` tags with img elements or others inside?
</blockquote></details>

Provided separate component for `titleWithIcon` on db pages level, as it used in  2 containers


<!-- Sent from Reviewable.io -->

dhartunian reviewed (changes requested):
@elkmaster this looks great! Can you add two storybook pages for databases and grants so we can ensure these designs stay consistent?

each should have at least 3 stories:
- with data
- with no tables
- in loading state

> Reviewed 2 of 10 files at r1, 11 of 11 files at r4.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47753#-:-M8QMcU61c-ul48aNo5V:b5wquzn)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dhartunian)



<!-- Sent from Reviewable.io -->

celiala reviewed (changes requested):
Thanks for integrating the design system styles - it's looking so nice!

The feedback I have is just for the PR/commit messages themselves:

* Can you update the PR title to say _what_ was done to the Database page (Based on the PR description, perhaps 'ui: update Database list styles to match design system'?)
* Similarly, could you update the release note, i.e. "Updated Admin UI database list styles to match design system." (see [Release Notes](https://wiki.crdb.io/wiki/spaces/CRDB/pages/186548364/Release+notes))
* I haven't reviewed the actual code changes too closely, but based on the commit messages, I think these 4 commits should be squashed into 1 commit? From [Split your change into logical chunks](https://wiki.crdb.io/wiki/spaces/CRDB/pages/181633464/Your+first+CockroachDB+PR) and [Pay Attention To](https://wiki.crdb.io/wiki/spaces/CRDB/pages/73072807/Git+Commit+Messages): 
> "Each commit should represent one logical change to the codebase with a descriptive message that explains the rationale for the change."... 
> "Do not include patch set-specific comments... [which] also applies to comments such as "Added unit tests", "Fixed localization problems", or any other such patch set to patch set changes that don't affect the overall intent of your commit.")
* Related to the point above, can you add more about what specifically changed to the commit description(s)? (see also [Commit description](https://wiki.crdb.io/wiki/spaces/CRDB/pages/73072807/Git+Commit+Messages#GitCommitMessages-Commitdescription))

Thanks!

dhartunian commented:
@elkmaster Grants page doesn't load for me. I get a blank page with:

```
Uncaught TypeError: this.props.refreshDatabaseDetails is not a function
    at DatabaseSummaryGrants.componentDidMount (index.tsx:91)
    at commitLifeCycles (react-dom.development.js:3719)
    at commitLayoutEffects (react-dom.development.js:4327)
    at HTMLUnknownElement.callCallback (react-dom.development.js:124)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:144)
    at invokeGuardedCallback (react-dom.development.js:159)
    at commitRootImpl (react-dom.development.js:4282)
    at unstable_runWithPriority (scheduler.development.js:701)
    at runWithPriority$2 (react-dom.development.js:2260)
    at commitRoot (react-dom.development.js:4245)
3react-dom.development.js:3705 The above error occurred in the <DatabaseSummaryGrants> component:
    in DatabaseSummaryGrants (created by ConnectFunction)
    in ConnectFunction (created by DatabaseGrantsList)
    in div (created by DatabaseGrantsList)
    in div (created by DatabaseGrantsList)
    in DatabaseGrantsList (created by ConnectFunction)
    in ConnectFunction (created by Context.Consumer)
    in withRouter(Connect(DatabaseGrantsList)) (created by Context.Consumer)
    in Route (created by App)
    in Switch (created by App)
    in div (created by Layout)
    in div (created by Layout)
    in div (created by Layout)
    in RequireLogin (created by ConnectFunction)
    in ConnectFunction (created by Context.Consumer)
    in withRouter(Connect(RequireLogin)) (created by Layout)
    in Layout (created by ConnectFunction)
    in ConnectFunction (created by Context.Consumer)
    in withRouter(Connect(Layout)) (created by App)
    in Route (created by App)
    in Switch (created by App)
    in Router (created by ConnectedRouter)
    in ConnectedRouter (created by Context.Consumer)
    in ConnectedRouterWithContext (created by ConnectFunction)
    in ConnectFunction (created by App)
    in Provider (created by App)
    in App

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://fb.me/react-error-boundaries to learn more about error boundaries.
```

Once nice outcome is that this *was* caught by the Storybook!

![Screen Shot 2020-06-12 at 1 16 34 PM](https://user-images.githubusercontent.com/986307/84529198-09afec80-acaf-11ea-96f7-59b0e1d7fbf5.png)

Can you add a unit test to ensure the component can render successfully?

elkmaster commented:
Fixed typo in mock data for storybook.
Added unit test coverage for both tables and all states.

dhartunian commented on pkg/ui/src/views/databases/containers/databaseGrants/index.tsx:
```diff
@@ -27,53 +30,60 @@ const grantsSortSetting = new LocalSetting<AdminUIState, SortSetting>(
 
+  noDatabaseResults = () => (
+    <>
+      <h3 className="table__no-results--title"><DatabaseIcon />This grands has no users.</h3>
```
Can you check designs or with Anne/Ryan about this text? It doesn't make sense as is to me.

Maybe something like "No grants on this table".

elkmaster replied:
done

vladlos committed:
ui: update Database page styles to match design system

Style updates of `database-tables` and `grants` tables.
Basic unit tests
Additional storybook items to display loading, empty and fulfilled states of updated components

Resolves: #45725

Release note(ui): Updated Admin UI database page styles to match design system.

celiala reviewed (approved):
Thanks for making those updates -- LGTM!

dhartunian reviewed (approved):
lgtm

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47753#-:-MCwGBoxBkLNwl7m0w7l:bb7k0qh)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dhartunian)



<!-- Sent from Reviewable.io -->

dhartunian commented:
bors r+
